components/rhel/misc/hyperv/rhel-k8s-singlenode-hyperv/Vagrantfile

```diff
@@ -0,0 +1,64 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+REQUIRED_PLUGINS = %w(vagrant-service-manager vagrant-registration vagrant-sshfs)
+errors = []
+
+def message(name)
+  "#{name} plugin is not installed, run `vagrant plugin install #{name}` to install it."
+end
+# Validate and collect error message if plugin is not installed
+REQUIRED_PLUGINS.each { |plugin| errors << message(plugin) unless Vagrant.has_plugin?(plugin) }
+unless errors.empty?
+  msg = errors.size > 1 ? "Errors: \n* #{errors.join("\n* ")}" : "Error: #{errors.first}"
+  fail Vagrant::Errors::VagrantError.new, msg
+end
+
+# Vagrantfile for single node k8s setup
+Vagrant.configure(2) do |config|
+  config.vm.box = "cdkv2"
+
+  if ENV.has_key?('SUB_USERNAME') && ENV.has_key?('SUB_PASSWORD')
+    config.registration.username = ENV['SUB_USERNAME']
+    config.registration.password = ENV['SUB_PASSWORD']
+  end
+
+  config.vm.provider "libvirt" do |libvirt, override|
+    libvirt.driver = "kvm"
+    libvirt.memory = 1024
+    libvirt.cpus = 2
+  end
+
+  config.vm.provider "virtualbox" do |vbox, override|
+    vbox.memory = 1024
+    vbox.cpus = 2
+
+    # Enable use of more than one virtual CPU in a virtual machine.
+    vbox.customize ["modifyvm", :id, "--ioapic", "on"]
+  end
+
+  # Fix for ADB #334
```

Comment:
This configuration is not a fix. This is the right way to do it.  IMO, we should add a comment like
"# Do not configure default OpenShift provider for CDK, more at ADB #334 "